app/components/search_result/results_metadata_component.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SearchResult
+  class ResultsMetadataComponent < ViewComponent::Base
+    def initialize(document:, template_fields: [])
```

Comment:
Do we need a default for template_fields?  Isn't it always required?
```suggestion
    def initialize(document:, template_fields:)
```